### PR TITLE
Add fixtures and helpers in e2e framework for BackendConfig

### PR DESCRIPTION
**Pull request by rramkumar1**
/assign @bowei

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/ingress-gce/331)

### CODE REVIEW

File: pkg/fuzz/helpers.go [hunk 1]
```diff
@@ -176,3 +177,63 @@ func (i *IngressBuilder) AddTLS(hosts []string, secretName string) *IngressBuild
...
+}
+
+// Build returns a constructed BackendConfig. The BackendConfig is a copy, so the Builder
+// can be reused to construct multiple BackendConfig definitions.
+func (b *BackendConfigBuilder) Build() *backendconfig.BackendConfig {
+	return b.backendConfig.DeepCopy()
+}
+
+// ToggleCDN enables or disables CDN on the BackendConfig.
+func (b *BackendConfigBuilder) ToggleCDN(enabled bool) *BackendConfigBuilder {
```
Comment (bowei) [inline pkg/fuzz/helpers.go]: EnableCDN

File: pkg/fuzz/helpers.go [hunk 2]
```diff
@@ -176,3 +177,63 @@ func (i *IngressBuilder) AddTLS(hosts []string, secretName string) *IngressBuild
...
+	if b.backendConfig.Spec.Cdn == nil {
```
Comment (bowei) [inline pkg/fuzz/helpers.go]: if enabled == false {
  b.backendConfig.Spec.Cdn = nil
}

Comment (rramkumar1) [reply pkg/fuzz/helpers.go]: I'm not following.. if enabled is false, I would want to set Cdn.Enabled to false, not set the pointer to nil.

File: pkg/fuzz/helpers.go (hunk 1 above)
Comment (rramkumar1) [reply pkg/fuzz/helpers.go]: Done.

**Pull request merged**